File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i=1;j=1;frequency=1
        while i<len(nums):
            if nums[i] != nums[i-1]:
                frequency=1;
                nums[j] = nums[i]
                j+=1
            elif nums[i] == nums[i-1]:
                if frequency <2:
                    frequency+=1
                    nums[j] = nums[i]
                    j+=1
            i+=1
        return j
        

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int i=1,j=1,frequency=1;
        while(i<nums.length){
            if(nums[i-1] != nums[i]){
                frequency=1;
                nums[j++] = nums[i];
            }else if(nums[i] == nums[i-1]){
                if(frequency < 2){
                    frequency++;
                    nums[j++] =nums[i];
                }
            }
            i++;
        }return j;
    }
}